inference/core/workflows/core_steps/classical_cv/distance_measurement/v1.py

```diff
@@ -0,0 +1,345 @@
+                },
+            },
+        },
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name=OUTPUT_KEY, kind=[DICTIONARY_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class DistanceMeasurementBlockV1(WorkflowBlock):
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        predictions: Batch[sv.Detections],
+        object_1_class_name: str,
+        object_2_class_name: str,
+        reference_axis: Literal["horizontal", "vertical"],
+        calibration_type: Literal["reference object", "pixel-ratio"],
+        reference_object_class_name: str,
+        reference_width: float,
+        reference_height: float,
+        pixel_ratio: float,
+    ) -> BlockResult:
+        if calibration_type == "reference object":
+            reference_predictions = predictions
+            distances = measure_distance_with_reference_object(
+                detections=predictions,
+                object_1_class_name=object_1_class_name,
+                object_2_class_name=object_2_class_name,
+                reference_predictions=reference_predictions,
+                reference_object_class_name=reference_object_class_name,
+                reference_width=reference_width,
+                reference_height=reference_height,
+                reference_axis=reference_axis,
+            )
+        elif calibration_type == "pixel-ratio":
+            distances = measure_distance_with_pixel_ratio(
+                detections=predictions,
+                pixel_ratio=pixel_ratio,
+                object_1_class_name=object_1_class_name,
+                object_2_class_name=object_2_class_name,
+                reference_axis=reference_axis,
+            )
+        else:
+            raise ValueError(f"Invalid calibration type: {calibration_type}")
+        
+        return {OUTPUT_KEY: distances}
+
+
+def measure_distance_with_reference_object(
+    detections: Batch[sv.Detections],
+    object_1_class_name: str,
+    object_2_class_name: str,
+    reference_predictions: sv.Detections,
+    reference_object_class_name: str,
+    reference_width: float,
+    reference_height: float,
+    reference_axis: Literal["horizontal", "vertical"],
+):# -> List[Dict[str, Union[str, float]]]:
+    reference_bbox_1= None
+    reference_bbox_2 = None
+        
+    reference_bbox_1, reference_bbox_2 = find_reference_bboxes(detections, object_1_class_name, object_2_class_name)
+
+    if not reference_bbox_1 or not reference_bbox_2:
+        raise ValueError(f"Reference class '{object_1_class_name}' or '{object_2_class_name}' not found in predictions.")
+
+    if has_overlap(reference_bbox_1, reference_bbox_2) or has_axis_overlap(reference_bbox_1, reference_bbox_2, reference_axis):
```

Comment:
I find this if statement hard to understand - especially the latter part.
The root source of my confusion is the choice to measure distance along single dimension - why not to calculate distance in 2D?

Reply:
I understand the confusion.

The goal is to measure the distance or gap between the closest edges of two bounding boxes on a 2D plane.

I'm adding an illustration to clarify this further:

<img width="1049" alt="image" src="https://github.com/user-attachments/assets/93c6deac-3a42-47f3-a0dc-ba6b519813d9">

- `has_overlap`: check if bboxes overlap.
- `has_axis_overlap`: check if there isn't a `vertical` or `horizontal` gap between the bboxes.